app/page.tsx

```diff
@@ -34,6 +35,10 @@ export default function Home() {
   const { authenticated } = usePrivy();
   const { currentStep, setCurrentStep } = useStep();
   const { selectedNetwork } = useNetwork();
+  
+  const router = useRouter();
+  const { LP: providerId } = router.query; 
+  const providerIdString = Array.isArray(providerId) ? providerId[0] : providerId;
```

Comment:
should we validate this? in case the provided lp value isn't parsable to a string then it should be an empty string `""`

Reply:
Made sure the the lp value is a string either empty or nonempty